js/ui/handler/touch_zoom_rotate.js

```diff
@@ -39,6 +39,13 @@ TouchZoomRotate.prototype = {
         this._startVec = p0.sub(p1);
         this._startScale = this._map.transform.scale;
         this._startBearing = this._map.transform.bearing;
+        this._scalingSignificantly = false;
+        this._rotatingSignificantly = false;
+        this._blockRotation = undefined;
```

Comment:
Yeah I had been thinking about that... Another option is to make it a state variable... something like this._gestureIntent which is then set to either "zoom" or "rotate"... that might be even clearer than just a boolean?